### PR TITLE
Feature/fireshield

### PR DESCRIPTION
**Pull request by TheSooth**

### CODE REVIEW

File: HydraSDKDemo/ViewController.swift [hunk 1]
```diff
@@ -105,76 +145,103 @@ class ViewController: UIViewController, CountryControllerProtocol {
...
     @IBAction func changeCountry(_ sender: UIButton) {
         let controller = self.storyboard?.instantiateViewController(withIdentifier: "CountryController") as! CountryController
         controller.currentCountry = self.country
         controller.delegate = self
         self.present(controller, animated: true, completion: nil)
     }
     
+    
+    @IBAction func toggleVPN(_ sender: UISwitch) {
+        if !self.isVpnConnected {
```
Comment (evgeniyd) [inline HydraSDKDemo/ViewController.swift]: shouldn't we protect from intermediate states here? e.g. do nothing if VPN is currently `.connecting` or `.disconnecting`.

Comment (evgeniyd) [reply HydraSDKDemo/ViewController.swift]: OK, from UX, I see that the switch is "locked" during these states.

File: HydraSDKDemo/ViewController.swift [hunk 2]
```diff
@@ -105,76 +145,103 @@ class ViewController: UIViewController, CountryControllerProtocol {
...
+            startVPN()
+        } else if isFireshieldEnabled {
+            presentDisableOptionsAlert()
+        } else {
+            stopVPN()
+        }
+    }
+    
+    private func presentDisableOptionsAlert() {
+        let alertController = UIAlertController(title: "VPN Connection",
```
Comment (evgeniyd) [inline HydraSDKDemo/ViewController.swift]: Nice UX with this alert!

File: HydraSDKDemo/ViewController.swift (hunk 1 above)
Comment (TheSooth) [reply HydraSDKDemo/ViewController.swift]: right :)

**Comment by evgeniyd** [comment]
Fireshield also has `AFVPNCategorizationDidChangeNotification` notification. Observing it allows the app in the foreground update UI with the resources being blocked.

**Pull request merged**